### PR TITLE
AdoptAWidget: Table (#69488)

### LINKED ISSUE #69488: Adopt a Widget - Table
The [Table](https://master-api.flutter.dev/flutter/widgets/Table-class.html) widget needs:

- A code / DartPad sample

For details about how to participate, see [flutter.dev/adoptawidget](https://flutter.dev/adoptawidget)

### PR DESCRIPTION
**Pull request by redsolver**
Added a Dartpad sample for `Table` with borders, multiple types of column widths and different vertical cell alignments.

This pull request is:

- [x] A code sample
- [ ] More references
- [ ] More explanation

closes #69488

### CODE REVIEW

File: packages/flutter/lib/src/widgets/table.dart [hunk 1]
```diff
@@ -81,6 +81,70 @@ class _TableElementRow {
 ///
 /// {@youtube 560 315 https://www.youtube.com/watch?v=_lbE0wsVZSw}
 ///
+/// {@tool dartpad --template=stateless_widget_material}
```
Comment (johnpryan) [inline packages/flutter/lib/src/widgets/table.dart:84]: I recommend using `--template=stateless_widget_scaffold` instead - it includes an AppBar and a white background.

File: packages/flutter/lib/src/widgets/table.dart [hunk 2]
```diff
@@ -81,6 +81,70 @@ class _TableElementRow {
...
+///             width: 128,
+///             color: Colors.purple,
+///           ),
+///           Container(
+///             height: 32,
+///             color: Colors.yellow,
+///           ),
+///           Center(
+///             child: Container(
+///               height: 64,
```
Comment (johnpryan) [inline packages/flutter/lib/src/widgets/table.dart:135]: nit: you could change the height to 32px to better show the `Center` widget in action.

**Review by johnpryan — APPROVED** [review]

File: packages/flutter/lib/src/widgets/table.dart (hunk 1 above)
Comment (redsolver) [reply packages/flutter/lib/src/widgets/table.dart]: Changed in https://github.com/flutter/flutter/pull/69588/commits/1678ebfa303062e075032c62c727d19c036998ed

File: packages/flutter/lib/src/widgets/table.dart [hunk 3]
```diff
@@ -81,6 +81,70 @@ class _TableElementRow {
...
+///             ),
+///           ),
+///         ],
+///       ),
+///     ],
+///   );
+/// }
+/// ```
+/// {@end-tool}
+///
```
Comment (johnpryan) [inline packages/flutter/lib/src/widgets/table.dart:147]: I think the build is failing due to a trailing character on this line:

```
/tmp/flutter sdk/packages/flutter/lib/src/widgets/table.dart:147: trailing U+0020 space character
```

**Review by johnpryan — CHANGES_REQUESTED** [review]
Looks good - I'm not sure why the build is failing.

File: packages/flutter/lib/src/widgets/table.dart (hunk 3 above)
Comment (redsolver) [reply packages/flutter/lib/src/widgets/table.dart]: Removed it

File: packages/flutter/lib/src/widgets/table.dart [hunk 4]
```diff
@@ -144,7 +144,7 @@ class _TableElementRow {
 /// }
 /// ```
 /// {@end-tool}
-/// 
+///
```
Comment (johnpryan) [inline packages/flutter/lib/src/widgets/table.dart:147]: ah, good catch.

**Comment by johnpryan** [comment]
@redsolver Thanks for participating in AdoptAWidget. We would like to send you a special thanks. If you are interested, please send an email to nyener at google dot com and provide a link to this issue.

**Pull request merged**